src/graphql/schema/node/createNodeFieldEntry.ts

```diff
@@ -46,7 +46,7 @@ export default function createNodeFieldEntry (buildToken: BuildToken): [string,
       if (!primaryKey || !primaryKey.read)
         throw new Error(`Invalid id, no readable primary key on collection named '${collection.name}'.`)
 
-      const value = await primaryKey.read(context, keyValue)
+      const value = await primaryKey.read(context, keyValue, resolveInfo, getGqlOutputType(buildToken, collection.type).gqlType)
```

Comment:
I was hoping you could help me out here! Also in a few places I've passed through the NotNull'd version of the type which I then have to strip in `getSelectFragment`. You've done an impressive job of writing very high quality code, but despite this because PostGraphQL is such a big project it does make a number of these things quite challenging to figure out - there's a lot going on!